compiler/modules/nback/module.bal

```diff
@@ -56,12 +57,42 @@ public function buildModule(bir:Module birMod, *Options options) returns [llvm:M
         DISubprogram? diFunc = di == () ? () : diFuncs[i];
         Scaffold scaffold = new(mod, llFuncs[i], diFunc, builder, defn, code);
         buildPrologue(builder, scaffold, defn.position);
-        check buildFunctionBody(builder, scaffold, code);
+        bir:Label[] sorted = [];
+        topoSort(code.blocks, 0, sorted);
+        int orphaned = code.blocks.length() - sorted.length();
+        if orphaned != 0 {
+            panic err:impossible(orphaned.toString() + " orphan basic block" + (orphaned == 1 ? "" : "s"));
+        }
+        check buildFunctionBody(builder, scaffold, code.blocks, sorted);
     }
     check birMod.finish();
     return [llMod, createTypeUsage(mod.usedSemTypes)];
 }
 
+function topoSort(bir:BasicBlock[] bbs, bir:Label l, bir:Label[] result) {
```

Comment:
What this is doing IIUC is adding to `result` the labels reachable in forward order from `l`. So I would have a name that reflects that. I would name variables a bit differently

* `blocks` rather than `bbs`
* `label` rather than `l`
* `reachableLabels` rather than `result`

Reply:
> reachableLabels rather than result


It calculates more than reachability. It is ordered such that each conditional appears before it's jump target. So I named it `ordered`.